daisi/src/manager/scenariofile_component.h

```diff
@@ -65,6 +65,26 @@ void serializeType(std::optional<T> &t, const std::string &key, YAML::Node node)
   }
 }
 
+/// @brief Deserialize data from node into variant with type T, if \p type matches
+/// T.typeName()
+/// @tparam T The type to parse if \p type matches \p req_type
+/// @param node the YAML node to read data from
+/// @param var variant to which the deserialized object is put
+/// @param type the actual type
+/// @return true if data was deserialzied
+template <typename T, typename... U>
+bool deserializeIf(YAML::Node node, std::variant<U...> &var, const std::string &type) {
+  if (type == T::typeName()) {
```

Comment:
How does this work? Does every T-class have to implement a static `typeName` method?

Reply:
Yes.
This avoids passing an additional string to this method every time it is called.  This also keeps the type name directly inside the structs itself to avoid scattering data around.